pkg/pushsync/pushsync.go

```diff
@@ -126,19 +130,35 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 		return swarm.ErrInvalidChunk
 	}
 
-	span, _, ctx := ps.tracer.StartSpanFromContext(ctx, "pushsync-handler", ps.logger, opentracing.Tag{Key: "address", Value: chunk.Address().String()})
-	defer span.Finish()
-
-	// Get price we charge for upstream peer read at headler
+	// Get price we charge for upstream peer read at headler.
 	responseHeaders := stream.ResponseHeaders()
 	price, err := headerutils.ParsePriceHeader(responseHeaders)
-
+	// if not found in returned header, compute the price we charge for this chunk.
 	if err != nil {
-		// if not found in returned header, compute the price we charge for this chunk and
-		ps.logger.Warningf("push sync: peer %v no price in previously issued response headers: %v", p.Address, err)
+		ps.logger.Warningf("pushsync: peer %v no price in previously issued response headers: %v", p.Address, err)
 		price = ps.pricer.PriceForPeer(p.Address, chunk.Address())
 	}
 
+	// if the peer is closer to the chunk, we were selected for replication. Return early.
+	if dcmp, _ := swarm.DistanceCmp(chunk.Address().Bytes(), p.Address.Bytes(), ps.address.Bytes()); dcmp == 1 {
+		_, err = ps.storer.Put(ctx, storage.ModePutSync, chunk)
+		if err != nil {
+			ps.logger.Errorf("pushsync: chunk store: %w", err)
```

Comment:
not sure if `%w` is the verb to use here... since you're logging and not returning w/ wrapping. probably `%v` should be used